.interface-validator.py

```diff
@@ -0,0 +1,48 @@
+
+import yaml
+import glob
+
+from enum import Enum
+from typing import List, Optional
+from pydantic import AnyHttpUrl, BaseModel, Field
+
+class StatusEnum(str, Enum):
+    draft = "draft"
+    live = "live"
+    retired = "retired"
+
+class CharmEntry(BaseModel):
+    name: str
+    url: AnyHttpUrl
+
+class Interface(BaseModel):
+    interface: str
+    version: int
+    internal: Optional[bool] = False
+    status: StatusEnum
+    requirers: List[CharmEntry]
+    providers: List[CharmEntry]
+
+
+class Validator:
+    def read_yaml(self, file_path: str) -> dict:
+        with open(file_path, 'r') as stream:
+            config = yaml.safe_load(stream)
+
+        return Interface(**config)
+
+    def get_files(self):
+        return glob.glob('./interfaces/**/interface.yaml', recursive=True)
+    
+
+    def run(self):
+        files = self.get_files()
+        print(f"Scanning {len(files)} interface definitions...")
+        for file in files:
+            try: 
```

Comment:
I'd do:

```python

errors = []
for file in files:
    try: 
        raw_interface = self.read_yaml(file)
    except yaml.errors.YAMLError:
        errors.append(f" - {file} contains invalid yaml")
    try: 
        Interface.validate(**raw_interface)
    except pydantic.ValidationError:
        errors.append(f" - {file} invalid interface schema: {e}")
        
if errors:
    exit("validation completed with errors: {'\n'.join(errors)}")
```

Reply:
this way you don't just get the first error in the stack, but you see all of them in case there's multiple